src/main/java/org/pesmypetcare/webservice/controller/petmanager/PetMedicationRestController.java

```diff
@@ -0,0 +1,132 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/medication")
+public class PetMedicationRestController {
+    @Autowired
+    private MedicationService medicationService;
+    /**
+     * Creates a Medication eaten by a pet on the database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName Date + name of the Medication
+     * @param Medication The Medication entity that contains the attributes of the Medication
+     */
+    @PostMapping("/{owner}/{petName}/{dateName}")
+    public void createMedication(@PathVariable String owner, @PathVariable String petName, @PathVariable String dateName,
+                           @RequestBody MedicationEntity Medication) {
+        medicationService.createMedication (owner, petName, dateName, Medication);
+    }
+
+    /**
+     * Deletes all the medication with the specified owner and pet from certain date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName date + name of the receipt of the medication
+     */
+    @DeleteMapping("/{owner}/{petName}/{dateName}")
+    void deleteByDateAndName(String owner, String petName, String dateName){
+        medicationService.deleteByDateAndName(owner, petName, dateName);
+    }
+
+    /**
+     * Deletes all the Medications of the specified pet from database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @DeleteMapping("/{owner}/{petName}")
+    public void deleteAllMedications(@PathVariable String owner, @PathVariable String petName)
+            throws DatabaseAccessException {
+        medicationService.deleteAllMedications(owner, petName);
+    }
+
+    /**
+     * Gets a Medication identified by its pet and date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName Date of the Medication
+     * @return The MedicationEntity identified by the data
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/{dateName}")
+    public MedicationEntity getMedicationData(@PathVariable String owner, @PathVariable String petName, @PathVariable String dateName)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 134).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)